### PR TITLE
Feature/issue004/add sem ver python script

### PR DESCRIPTION
**Pull request by noamgu**

### CODE REVIEW

File: script/sem-ver-action.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+#  imports
+import os
+import git
+from git import Repo
+
+#  variables
+script_path = os.getcwd()
+remote_url = "git@github.com:shiri2001/training.git"
+branch = "main"  # noqa:E501 change later to main so it only runs on main
+repo = Repo(script_path)
```
Comment (noamgu) [inline script/sem-ver-action.py:10]: The General Variables section is a good idea. It gives your reviewer/maintainer a single place  adding additional features, or fixing the code to make any necessary changes (such as variable updates).
Everything make sense except the `repo` why you declared it here?

File: script/sem-ver-action.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+#  clone repository from github if needed
+def clone():
+    try:
+        Repo.clone_from(remote_url, script_path)
+    except git.exc.GitCommandError:
+        pass
```
Comment (noamgu) [inline script/sem-ver-action.py:18]: If we catch error do your really want to just ignore 'pass' it? 😱

File: script/sem-ver-action.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def main():
+    # logging bot into github using desired branch
```
Comment (noamgu) [inline script/sem-ver-action.py:22]: The main method should be as clean as possible. All code blocks related to business logic should be placed in dedicated functions or classes.

File: script/sem-ver-action.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#  imports
```
Comment (noamgu) [inline script/sem-ver-action.py:1]: Please add a logger to the relevant places. Make sure you are using the appropriate log levels (info, debug, and error) where necessary.

**Review by noamgu — COMMENTED** [review]
@shiri2001  Since I'm the one who opened the PR, I can't "Request changes." So please consider this comment as is :)

File: script/sem-ver-action.py (hunk 3 above)
Comment (shiri2001) [reply script/sem-ver-action.py:22]: is there a particular part that should be moved?

File: script/sem-ver-action.py (hunk 4 above)
Comment (shiri2001) [reply script/sem-ver-action.py:1]: added!

File: script/sem-ver-action.py (hunk 3 above)
Comment (noamgu) [reply script/sem-ver-action.py:24]: @shiri2001 Excellent, now the main method is clean.
Let's tune this up.
Watch this -->
https://www.youtube.com/watch?v=pTB30aXS77U&ab_channel=ArjanCodes

**Pull request merged**